Blog nielx: add post Haiku Developer Tools Update (October 2023) (#658)
haiku/website: content/blog/nielx/2023-10-28_haiku_developer_tools_update_october_2023.md

+++
type = "blog"
title = "Haiku Developer Tools Update (October 2023)"
author = "nielx"
date = "2023-10-28 10:35:19+01:00"
tags = ["haiku", "software"]
+++

> This article lists some of the recent updates for developer tools that are available in the HaikuPorts applications. Many of these can be installed the HaikuDepot.

Reviewer: ```suggestion
This article lists some of the recent updates for developer tools that are available in the HaikuPorts repository. Many of these can be installed from HaikuDepot.
```

This is the first article on this topic, but it may become a regular (quarterly?) series. Let me know in the comments what you find of these notes, and what else you think should be covered.

<!--more-->

## GCC 13.2.0

> The Haiku nightlies are now built with GCC 13.2.0 as the primary compiler and it is available for software developers. It can also be used on a Haiku R1 beta 4 system. The previous version was GCC 11.3.0. The packages support **C, C++, and Fortran.**

Reviewer: ```suggestion
The Haiku nightlies are now built with GCC 13.2.0 as the primary compiler (the previous version was GCC 11.3.0) and it is available for software developers. It can also be used on a Haiku R1 beta 4 system. The packages support **C, C++, and Fortran.**
```

The updated version includes all improvements in the [GCC 12](https://gcc.gnu.org/gcc-12/changes.html) and [GCC 13](https://gcc.gnu.org/gcc-13/changes.html) builds. Since Haiku's primary language is C++, it is particularly notable that there is now improved support for C++20 features, as well as some more early support for C++23.

Check out the packages on **HaikuDepot**:

> - `gcc`: [64 bit](https://depot.haiku-os.org/#!/pkg/gcc/haikuports/haikuports_x86_64/13/2/0_2023_08_10/-/3/x86_64) | [32 bit](https://depot.haiku-os.org/#!/pkg/gcc_x86/haikuports/haikuports_x86_gcc2/13/2/0_2023_08_10/-/3/x86_gcc2)

Reviewer 2: And below

> - `gcc`: [64 bit](https://depot.haiku-os.org/#!/pkg/gcc/haikuports/haikuports_x86_64/13/2/0_2023_08_10/-/3/x86_64) | [32 bit](https://depot.haiku-os.org/#!/pkg/gcc_x86/haikuports/haikuports_x86_gcc2/13/2/0_2023_08_10/-/3/x86_gcc2)

Author: I think I prefer to link to the specific version for this article, but thanks for pointing out that there are other direct links!

- `llvm17_python310`: [64 bit](https://depot.haiku-os.org/#!/pkg/llvm17_python310/haikuports/haikuports_x86_64/17/0/1/-/3/x86_64) | [32 bit](https://depot.haiku-os.org/#!/pkg/llvm17_x86_python310/haikuports/haikuports_x86_gcc2/17/0/1/-/3/x86_gcc2)

## Python 3.10, 3.11 & 3.12

The Python interpreter has been updated recently as well, and the standard python interpreter on Haiku is now Python 3.10.

Check out the packages on **HaikuDepot**:

- `python3.10`: [64 bit](https://depot.haiku-os.org/#!/pkg/python3.10/haikuports/haikuports_x86_64/3/10/13/-/1/x86_64) | [32 bit](https://depot.haiku-os.org/#!/pkg/python3.10_x86/haikuports/haikuports_x86_gcc2/3/10/13/-/1/x86_gcc2)

- `python 3.11`: [64 bit](https://depot.haiku-os.org/#!/pkg/python3.11/haikuports/haikuports_x86_64/3/11/6/-/1/x86_64) | [32 bit](https://depot.haiku-os.org/#!/pkg/python3.11_x86/haikuports/haikuports_x86_gcc2/3/11/6/-/1/x86_gcc2)

- `python 3.12`: [64 bit](https://depot.haiku-os.org/#!/pkg/python3.12/haikuports/haikuports_x86_64/3/12/0/-/1/x86_64?bcguid=bc2938-FWTK) | [32 bit](https://depot.haiku-os.org/#!/pkg/python3.12_x86/haikuports/haikuports_x86_gcc2/3/12/0/-/1/x86_gcc2)

## Qt Creator 11.0.3

> This IDE from the makers of Qt supports full-fledged C++ development, including code completion using `clangd` (part of the LLVM package). See what's new in [version 11](https://www.qt.io/blog/qt-creator-11-released) . Previous release on Haiku was 6.0.1.

Reviewer: ```suggestion
This IDE from the makers of Qt supports full-fledged C++ development, including code completion using `clangd` (part of the LLVM package). See what's new in [version 11](https://www.qt.io/blog/qt-creator-11-released). Previous release on Haiku was 6.0.1.
```

Check out the package on **HaikuDepot**: [64 bit](https://depot.haiku-os.org/#!/pkg/qt_creator/haikuports/haikuports_x86_64/11/0/3/-/2/x86_64) | [32 bit](https://depot.haiku-os.org/#!/pkg/qt_creator_x86/haikuports/haikuports_x86_gcc2/11/0/3/-/2/x86_gcc2)

## CudaText 1.200.0.0

> This modern text editor for developers is written in Object Pascal, is fast, and has a lot of features that are available as plugins, including code completion using `clangd`(part of the LLVM package). See the [change log](https://github.com/Alexey-T/CudaText/blob/1.200.0/app/readme/history.txt) to see what's new.

Reviewer: ```suggestion
This modern text editor for developers is written in Object Pascal, is fast, and has a lot of features that are available as plugins, including code completion using `clangd` (part of the LLVM package). See the [change log](https://github.com/Alexey-T/CudaText/blob/1.200.0/app/readme/history.txt) to see what's new.
```
